flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterEntrypoint.java

```diff
@@ -151,6 +153,19 @@ protected ClusterEntrypoint(Configuration configuration) {
         this.configuration = generateClusterConfiguration(configuration);
         this.terminationFuture = new CompletableFuture<>();
 
+        if (configuration.get(JobManagerOptions.SCHEDULER_EXECUTION_MODE)
+                        == SchedulerExecutionMode.REACTIVE
+                && !this.getClass()
+                        .getSimpleName()
+                        .equals("StandaloneApplicationClusterEntryPoint")) {
```

Comment:
let's add `protected boolean supportsReactiveMode() { return false; }` and override it in the `StandaloneApplicationClusterEntryPoint`
adjust the test accordingly to use a test entrypoint implementation.